nilearn/reporting/_utils.py

```diff
@@ -49,23 +49,19 @@ def model_attributes_to_dataframe(model):
     attributes_df: pandas.DataFrame
         DataFrame with the pertinent attributes of the model.
     """
-    attributes_df = OrderedDict(
-        (
-            attr_name,
-            (
-                str(getattr(model, attr_name))
-                if isinstance(getattr(model, attr_name), dict)
-                else getattr(model, attr_name)
-            ),
-        )
-        for attr_name in model.get_params()
-    )
-
-    for k, v in attributes_df.items():
+    attributes = []
+    for attr_name in model.get_params():
+        if isinstance(getattr(model, attr_name), dict):
+            attributes.append((attr_name, str(getattr(model, attr_name))))
+        elif getattr(model, attr_name) is not None:
+            attributes.append((attr_name, getattr(model, attr_name)))
+    attributes = OrderedDict(attributes)
+
+    for k, v in attributes.items():
```

Comment:
There is also `reporting.glm_reporter._glm_model_attributes_to_dataframe`. Could they be merged with `model_attributes_to_dataframe`?